File: tpl/src/rngsse2/examples/examples_CPU_C/gq58x1.c

#include<stdio.h>
#include<gq58x1.h>

#define NN       100000000UL

int main(void){ 
   long i; unsigned int sum=0;
   gq58x1_state state;
   gq58x1_init_(&state);
   gq58x1_print_state_(&state);
   for(i=0;i<NN;i++) sum+=gq58x1_generate_(&state);
   printf("%ld GQ58X1 pseudorandom numbers generated.\n",NN);
   printf("Fractional part of the total sum of generated numbers: %f\n",sum/4294967296.);
   printf("Next output value: %f\n",gq58x1_generate_(&state)/4294967296.);
   return 0;
}

File: tpl/src/rngsse2/examples/examples_CPU_FORTRAN/gq58x1.f90

PROGRAM GQ58X1_Example

   PARAMETER (N=100000000)
   INTEGER i,GQ58X1_generate,sum
   REAL val,rsum

   TYPE GQ58X1_state
     INTEGER z(132)
   END TYPE GQ58X1_state
   TYPE(GQ58X1_state) state

   CALL GQ58X1_init ( state )
   CALL GQ58X1_Print_State (state)

   sum = 0

   DO i=1,N
     sum = sum + GQ58X1_generate (state)
   END DO

   WRITE(*,1) N
   rsum = sum/4294967296.
   val = GQ58X1_generate (state)/4294967296.
   IF (rsum<0) THEN
     rsum=rsum+1
   ENDIF
   IF (val<0) THEN
     val=val+1
   ENDIF
   WRITE(*,2) rsum
   WRITE(*,3) val

1  FORMAT(I9," GQ58X1 pseudorandom numbers generated.")
2  FORMAT("Fractional part of the total sum of generated numbers: ",F8.6)
3  FORMAT("Next output value: ",F8.6)

END PROGRAM GQ58X1_Example

